### DISCUSSION TITLE
GoogleAnalytics SDK in AdobeAIRMainActivity

### DISCUSSION THREAD

**Discussion by Klug76** [airsdk/Adobe-Runtime-Support#1593] [General]
Google Play Developer Console requires publishers to declare if they have ads or not. My app does not have ads, but flagged as having the GoogleAnalytics SDK.
Here i found that the AdobeAIRMainActivity class uses GoogleAnalytics::Tracker("UA-54849355-1"):
[https://forum.starling-framework.org/d/21074-adobe-air-adobeairmainactivity-and-google-analytics/12](https://forum.starling-framework.org/d/21074-adobe-air-adobeairmainactivity-and-google-analytics/12)
Any chance to remove google analytics from AIR (at least from the paid version)?

**Comment by ajwfrost** [comment]
Hi .. that class, and all the google analytics, look like they're only included within the old 'shared' AIR runtime application that Adobe used to publish separately. 

I can't see anything else in the source that looks like it includes the Google Analytics libraries .. 

Are you able to open your package within Android Studio via Build | Analyse APK, and see if you can find the analytics classes within the list shown when you view the classes.dex file?

thanks

**Comment by Klug76** [comment]
![google_in_air](https://user-images.githubusercontent.com/1420061/150784020-b8908e66-b591-4b4a-a6c0-e1dd46b84045.png)
It looks like this class is used in the apk/aab assembly: `` SDK_Air330/lib/android/lib/runtimeClasses.jar/com/adobe/air/AdobeAIRMainActivity.class``

**Reply by ajwfrost** [reply]
Okay good spot ! I've just gone through again, slower, and found that the bundle does have that AdobeAIRMainActivity in it...
Not sure why this is coming through, we'll check back through the usages and try to find why it's being included!

thanks

**Reply by Klug76** [reply]
Thanks! By the way, this class is included even in an empty project

**Reply by ajwfrost** [reply]
Thanks, yes it looks like the actual functionality is commented out but the classes are then incorporated anyway (a fair amount of reflection being used which may make it be over-cautious about excluding things). So I don't believe any of that is actually active, we should do some clean-up though to remove all the irrelevant/unused classes and properties and will check that this resolves it...

thanks

**Reply by JeroenWimmers** [reply]
Hi Andrew, is there any update on this? We're facing the same issue

**Reply by ajwfrost** [reply]
We've got updates in our next SDK that remove those classes but we have found a few issues during release testing so it's a bit delayed..
It might be that you're able to extract the below zip into the "lib/android/lib" folder of the latest released SDK, which should sort it out without introducing the other problems that we're currently working to resolve!

thanks
[runtimeClasses.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/7977842/runtimeClasses.zip)

**Reply by JeroenWimmers** [reply]
Thank you! We'll try it out
